crates/experimentation_platform/src/api/experiments/handlers.rs

```diff
@@ -548,11 +549,33 @@ async fn list_experiments(
         return Ok(HttpResponse::NotModified().finish());
     };
 
-    let query_builder = |filters: &ExpListFilters| {
+    let query_builder = |filters: &ExperimentListFilters| {
         let mut builder = experiments::experiments.into_boxed();
-        if let Some(states) = filters.status.clone() {
+        if let Some(ref states) = filters.status {
             builder = builder.filter(experiments::status.eq_any(states.0.clone()));
         }
+        if let Some(ref experiment_name) = filters.experiment_name {
+            builder =
+                builder.filter(experiments::name.like(format!("%{}%", experiment_name)));
+        }
+        if let Some(ref context_search) = filters.context {
+            builder = builder.filter(
+                sql::<Bool>("context::text LIKE ")
+                    .bind::<Text, _>(format!("%{}%", context_search)),
+            );
+        }
+        if let Some(ref created_by) = filters.created_by {
+            builder =
+                builder.filter(experiments::created_by.eq_any(created_by.0.clone()));
+        }
+        if let Some(experiment_ids) = filters.experiment_ids.clone() {
+            let experiment_ids: HashSet<i64> = experiment_ids
+                .0
+                .iter()
+                .map(|i| i.parse::<i64>().unwrap_or_default())
```

Comment:
shouldn't we remove those experiment ids which are not proper integers, instead of searching for `0` ?
```suggestion
                .filter_map(|i| i.parse::<i64>())
```

Reply:
Yeah your suggestion makes way more sense. Will make this change